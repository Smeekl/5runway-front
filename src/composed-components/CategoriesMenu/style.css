.categories-menu {
  display: flex;
  justify-content: space-between;
}

.categories-menu__discount {
  margin-right: 180px;
  position: relative;
}

.discount__block {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.categories-menu__btn {
  flex-direction: column;
  display: flex;
  justify-content: space-between;
}

.discount__area {
  padding: 20px 20px 20px 20px;
  right: -179px;
  top: 115px;
  border-radius: 20px;
  width: 228px;
  height: 174px;
  position: absolute;
  flex-direction: row;
  background: var(--color-primary);
}
.q {
  font-weight: 800;
  font-size: 36px;
  color: var(--color-primary-blue);
  margin: 0;
}
.qq {
  font-weight: 600;
  font-size: 20px;
  line-height: 24px;
  margin: 0;
  color: var(--color-secondary-white);
}

.discount__sell-now__btn {
  color: var(--color-secondary-white);
  width: 343px;
  height: 48px;
  background: var(--color-primary-blue);
  border-radius: 15px;
  transition: 0.3s all;
  z-index: 999999;
}
.discount__sell-now {
  font-weight: 700;
  font-size: 40px;
}
.discount__sell-now__description {
  width: 195px;
  font-weight: 300;
  font-size: 18px;
}
