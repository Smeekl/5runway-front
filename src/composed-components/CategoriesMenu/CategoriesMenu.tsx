import { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';
import axios from 'axios';
import { ReactComponent as IconPlane } from '@svgIcons/iconPlane.svg';
import { CategoriesMenuCard } from '@components/CategoriesMenuCard/CategoriesMenuCard';
import Button from '@components/Button/Button';
import './style.css';

export const CategoriesMenu = () => {
  const [categories, setCategories] = useState([]);

  useEffect(() => {
    axios
      .get('https://62c80e240f32635590d045ac.mockapi.io/categories-menu')
      .then((res) => setCategories(res.data));
  }, []);

  return (
    <>
      <div className="categories-menu">
        {categories.map(({ name, imgUrl, route }, index) => (
          <Link to={route} key={index}>
            <CategoriesMenuCard name={name} imgUrl={imgUrl} />
          </Link>
        ))}
      </div>
      <div className="discount__block ">
        <div className="categories-menu__btn">
          <p className="discount__sell-now">Sell Now</p>
          <p className="discount__sell-now__description">
            You can place your ad for aviation equipment
          </p>
          <Button className="discount__sell-now__btn">Sell Now </Button>
        </div>
        <div className="categories-menu__discount">
          <IconPlane />
          <div className="discount__area">
            <p className="q">-50%</p>
            <p className="qq">have time to buy a subscription</p>
          </div>
        </div>
      </div>
    </>
  );
};
