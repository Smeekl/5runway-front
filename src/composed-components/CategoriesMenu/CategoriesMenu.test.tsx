import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { CategoriesMenu } from './CategoriesMenu';

describe('CategoriesMenu UI tests', () => {
  test('should render correctly', () => {
    const { container } = render(
      <MemoryRouter>
        <CategoriesMenu />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });

  test('should be in the document', () => {
    const { container } = render(
      <MemoryRouter>
        <CategoriesMenu />
      </MemoryRouter>
    );
    expect(container).toBeInTheDocument();
  });
});
