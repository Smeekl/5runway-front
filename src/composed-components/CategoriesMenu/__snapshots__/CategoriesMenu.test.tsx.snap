// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Conditions UI tests should render correctly 1`] = `
<div>
  <div
    class="categories-menu"
  />
  <div
    class="discount__block "
  >
    <div
      class="categories-menu__btn"
    >
      <p
        class="discount__sell-now"
      >
        Sell Now
      </p>
      <p
        class="discount__sell-now__description"
      >
        You can place your ad for aviation equipment
      </p>
      <button
        class="ant-btn ant-btn-default discount__sell-now__btn"
        type="button"
      >
        <span>
          Sell Now 
        </span>
      </button>
    </div>
    <div
      class="categories-menu__discount"
    >
      <reactcomponent />
      <div
        class="discount__area"
      >
        <p
          class="q"
        >
          -50%
        </p>
        <p
          class="qq"
        >
          have time to buy a subscription
        </p>
      </div>
    </div>
  </div>
</div>
`;
